src/client/client/game.js

```diff
@@ -139,6 +150,121 @@ let game = (function() {
 				_animateRoundTimer();
 				_DOMElements.state.innerHTML = "Race started!";
 				break;
+
+			// The race has finished
+			case "finished":
+				if (!isInitialState) {
+					_audio.end.play();
+
+					_DOMElements.raceLeaderboard.className = "visible";
+
+					_enteredMarbleList.sort((a, b) => {
+						if (a.finished && b.finished) {
+							return b.time - a.time;
+						} else if (a.finished) {
+							return 1;
+						} else if (b.finished) {
+							return -1;
+						} else {
+							return 0;
+						}
+					});
+
+					// Reverse it cause im stupid
+					_enteredMarbleList.reverse();
```

Comment:
Why not flip the conditions in the sorting function instead? ~~That way you can remove that comment and no one will know what really happened.~~

Reply:
~~cause im stupid~~